src/vm/node_modules/VM.js

```diff
@@ -17641,11 +17641,21 @@ function doShutdownStop(vmobj, options, callback)
                 cb();
             });
         }, function waitInstalled(cb) {
-            VM.waitForZoneState(vmobj, 'installed', {timeout: STOP_TIMEOUT},
-                function shutdownWaitCb(err) {
+            var stop_timeout = Date.now(0) + STOP_TIMEOUT * 1000;
+            var ival = setInterval(function () {
+                // Every second, tell `systemd-shutdown` to re-check
+                // its list of remaining processes.
+                // (By default, `systemd-shutdown` waits 90 seconds
+                // for a root cgroup SIGCHLD that is never sent.)
+                fs.readlink('/proc/' + vmobj.pid + '/path/a.out',
```

Comment:
Oh yes... I forgot to look at the bigger picture.  You know the pid already.

I had the man page section wrong, it's proc(5) now.  In illumos /proc/PID/argv, you get the information you need, albeit in a form that requires C-like parsing (strings with '\0' terminators).

Reply:
DAMN... I'm so sorry... this works exactly as it should for you.  This is an improvement.